### ISSUE TITLE
How to import RSA private key through pcap?

### ISSUE THREAD

**Issue by junhui** [mranney/node_pcap#158]
i'm going to use node_pcap analysis http traffic which include https, i know something like wireshark have  way to import the RSA key, is there any way i can use node_pcap+libpcap to do the samething?

**Comment by jmaxxz** [comment]
No, you are either going to have to do your own decryption. Or use a tool like ssldump to do the decryption first, though this will not result in a file usable by node_pcap.

**Comment by mranney** [comment]
If you want to contribute an implementation of decryption, I think that'd be great. :)

**Comment by jmaxxz** [comment]
^this

**Comment by junhui** [comment]
will try to see what i can do

**Comment by Domiii** [comment]
I have also looked into this, and unless there is someone who is already familiar with (or has a lot of time for learning) the basics of SSL, this is a rather daunting task.

For starters, Node's own [`ssl`](https://github.com/joyent/node/tree/master/src/node_crypto.cc) and [`tls`](https://github.com/joyent/node/tree/master/src/tls_wrap.cc) modules both rely on [openssl](https://github.com/joyent/node/tree/master/deps/openssl) and these [`SecureContext` native bindings](https://github.com/joyent/node/tree/master/src/node_crypto.cc#L266).

SSL programming with openssl is explained [here](http://h71000.www7.hp.com/doc/83final/ba554_90007/ch04s03.html). On the client side, it boils down to a connection being managed by openssl itself (established through `SSL_connect`). Not sure, if one can easily detach the transport from the security layer. Maybe a sort of piping mechanism from `pcap` to `openssl` would make things easier. Also note that whatever the mechanism, decryption must be handled before filtering.
